src/calendaragendamodel.h

```diff
@@ -67,49 +60,39 @@ class CalendarAgendaModel : public QAbstractListModel, public QQmlParserStatus
     Q_DECLARE_FLAGS(FilterModes, FilterMode)
     Q_FLAG(FilterModes)
 
-    explicit CalendarAgendaModel(QObject *parent = 0);
+    CalendarAgendaModel(QObject *parent = 0);
     virtual ~CalendarAgendaModel();
 
+    virtual QHash<int, QByteArray> roleNames() const;
+
     QDate startDate() const;
     void setStartDate(const QDate &startDate);
 
     QDate endDate() const;
     void setEndDate(const QDate &endDate);
 
-    int count() const;
-
     int filterMode() const;
     void setFilterMode(int mode);
 
     // CalendarAgendaModel takes ownership of the CalendarEventOccurrence objects
     void doRefresh(QList<CalendarEventOccurrence *>);
 
-    int rowCount(const QModelIndex &index) const;
-    QVariant data(const QModelIndex &index, int role) const;
-    Q_INVOKABLE QVariant get(int index, int role) const;
+    Q_INVOKABLE virtual QVariant get(int index, int role) const;
```

Comment:
Optionally could be marked override, some other methods too. Doesn't matter much.